timesketch/lib/analyzers/base_sessionizer_test.py

```diff
@@ -27,100 +25,96 @@ def test_analyzer(self):
+                               (self.analyzer_class.max_time_diff_micros / 2)
+                           ])
+
+        message = analyzer.run()
+        self.assertEqual(
+            message, 'Sessionizing completed, number of session created: 2')
+
+        event1 = (ds.get_event('test_index', '0', stored_events=True))
+        self.assertEqual(event1['_source']['session_id'],
+                         {analyzer.session_type: 1})
+
+        event2 = (ds.get_event('test_index', '101', stored_events=True))
+        self.assertEqual(event2['_source']['session_id'],
+                         {analyzer.session_type: 2})
+        self._check_surrounding_events(ds, [101], analyzer.session_type)
 
     @mock.patch('timesketch.lib.analyzers.interface.ElasticsearchDataStore',
                 MockDataStore)
     def test_edge_time_diff(self):
         """Test events with the edge time difference between them are
         allocated correctly."""
-        with mock.patch.object(
-                self.analyzer_class,
-                'event_stream',
-                return_value=_create_mock_event(
-                    0,
-                    2,
-                    time_diffs=[self.analyzer_class.max_time_diff_micros])):
-            index = 'test_index'
-            sketch_id = 1
-            analyzer = self.analyzer_class(index, sketch_id)
-            message = analyzer.run()
-            self.assertEqual(
-                message,
-                'Sessionizing completed, number of session created: 1')
-
-            ds = MockDataStore('test', 0)
-            event1 = (ds.get_event('test_index', '0', stored_events=True))
-            self.assertEqual(event1['_source']['session_id'],
-                             {analyzer.session_type: 1})
-            event2 = (ds.get_event('test_index', '101', stored_events=True))
-            self.assertEqual(event2['_source']['session_id'],
-                             {analyzer.session_type: 1})
-
-    def _check_surrounding_events(self, threshold_ids, session_type):
+        index = 'test_index'
+        sketch_id = 1
+        analyzer = self.analyzer_class(index, sketch_id)
+        analyzer.datastore.client = mock.Mock()
+        ds = analyzer.datastore
+
+        _create_mock_event(
+            ds, 0, 2, time_diffs=[self.analyzer_class.max_time_diff_micros])
+
+        message = analyzer.run()
+        self.assertEqual(
+            message, 'Sessionizing completed, number of session created: 1')
+
+        event1 = (ds.get_event('test_index', '0', stored_events=True))
+        self.assertEqual(event1['_source']['session_id'],
+                         {analyzer.session_type: 1})
+        event2 = (ds.get_event('test_index', '101', stored_events=True))
+        self.assertEqual(event2['_source']['session_id'],
+                         {analyzer.session_type: 1})
+
+    def _check_surrounding_events(self, ds, threshold_ids, session_type):
```

Comment:
The MockDataStore instance used here is now the same as the one used in the analyzer.